### ISSUE TITLE
Store boolean variables' refined annotations

### ISSUE THREAD

**Issue by byd110** [eisop/checker-framework#1407]
Some checkers apply special refinement on certain values, which often results in diverse stores during the analysis over conditionals, e.g. `obj == null` would refine `obj` to be `Nullable` in then branch and `NonNull` in else branch. However, the control flow analysis currently cannot handle the case if we wrap this boolean expression into a boolean variable. Consider the example below.

``` java
void inlinedCondition(@Nullable Object obj) {
    if(obj != null){
        obj.toString();
    }
}

void wrappedCondition(@Nullable Object obj) {
    boolean cond = obj != null;
    if(cond){
        obj.toString();
    }
}
```
Both call on `toString()` should be valid since `obj` is nonull in the then branch. However, there is an error of dereference of nullable thrown in `wrappedCondition`.

We can consider if we should have another store to carry such boolean variables conditional stores to resolve this problem.
